### ISSUE TITLE
[Global Calculated Coordinates/Waypoints] Remove old code

### ISSUE THREAD

**Issue by eddiemuc** [cgeo/cgeo#12380]
After switching to new global waypoints is done, old local coordinate codebase can be removed (excluding parts which are needed e.g. to read old Personal Notes and to migrate).

This should not be done before the new implementation was out in the field for a while to prove it is stable.

Following classes/files are candidates for removal:
```
cgeo.geocaching.ui.EditButton
cgeo.geocaching.ui.CalculateButton
cgeo.geocaching.ui.CalculatorVariable
cgeo.geocaching.models.CalcState
cgeo.geocaching.ui.dialog.CoordinatesCalculateDialog
coordinatescalculate_dialog.xml (both layout variants)
coordinates_calculate.xml
coordinate_equation.xml
strings.xml: values marked with "legacy"
```

**Comment by eddiemuc** [comment]
Took a look at this one today (removal of old Calculated Coordinate code). 

There is a large bunch of legacy code around which is still needed to parse the "old" format for calculated coordinates stored in personal note (the one with the "F-PLAIN"-marker created by @murggel  back in the days). If I remove this, then coordinates stored in personal note with this format can't be parsed again by c:geo. New calculated coordinates will be stored in a different format.

New format (used as of now): 
```
@Markierung 1 (W) {CC|N53 33.AAA|E009 59.BCD} "example_1"
$A=1 | $B=2 | $C=3 | $D=4
```
Old format (using `F-PLAIN`, no longer parseable if legacy code is removed): 
`@Markierung 1 (W) (F-PLAIN) N 53° 33.AAA E 009° 59.BCD' |A=1|B=2|C=3|D=4| "example_1"`

Question: (when) can we remove the parsing of F-PLAIN coordinates out or personal note?

**Comment by eddiemuc** [comment]
Note for self: most classes of package calculator can be removed if F-PLAIN does not need to be supported:
![image](https://user-images.githubusercontent.com/6909759/170819854-d40c5f4f-58ee-43c6-b8f6-30a646eea25d.png)

Exceptions/Additions:
* CalculatedCoordinateMigrator needs to stay
* FormulaParser might need adaptions (is also used for "current" format parsing
* WaypointParser needs adaptions (might still use FormulaParser or something else)

**Comment by murggel** [comment]
Probably we can make an announcement, that the migration of the F-Plain format will be removed in one of the next versions, so those who have still the format in some PN can do the migration before it will be removed.

**Comment by eddiemuc** [comment]
Since the current productive version still contains the old calculator logic, this should not be postponed now, but only after current beta has become the productive version (and maybe runs stable for some weeks).

Thus for now I will postpone working on this issue.

**Comment by ztNFny** [comment]
> An announcement on FB might be a good idea

Why an announcement on Facebook? This will miss 99% of users. I don't know a single user following that - me included.

**Comment by eddiemuc** [comment]
I looked at the code a bit more deeply. I think it should be possible to extract the core parts of F-PLAIN coordinate scanning and parse it as "plain" new calculated coordinates behind thje scenes without having to keep the whole class structure. 

So I am positive we can keep scanning the then-legacy F-PLAIN coordinates too. No need for an announcement then.

**Comment by fm-sys** [comment]
Just a small ping, Global Calculated Coordinates are released since quite some time meanwhile, so we should go ahead with doing the corresponding cleanup.

**Comment by eddiemuc** [comment]
> Just a small ping, Global Calculated Coordinates are released since quite some time meanwhile, so we should go ahead with doing the corresponding cleanup.

Absolutely! I will unassign myself in case you want to dig in right away

**Comment by fm-sys** [comment]
> > Just a small ping, Global Calculated Coordinates are released since quite some time meanwhile, so we should go ahead with doing the corresponding cleanup.
> 
> Absolutely! I will unassign myself in case you want to dig in right away

I'm unfortunately quite busy as well currently... I'll add label "prio - medium" for now

**Comment by eddiemuc** [comment]
PR #13656 is first part for this. It contains alternative parsing code for legacy F-PLAIN coordinates in user note

**Comment by eddiemuc** [comment]
PR #13703 will remove the legacy code
